Plan/sponge/src/main/java/com/djrapitops/plan/PlanSponge.java

```diff
@@ -223,21 +204,15 @@ public ColorScheme getColorScheme() {
 
     @Override
     public void registerCommand(Subcommand command) {
-        if (command == null) {
-            logger.warn("Attempted to register a null command!");
-            return;
-        }
-        for (String name : command.getAliases()) {
-            CommandManager commandManager = Sponge.getCommandManager();
-
-            CommandMapping registered = commands.get(name);
-            if (registered != null) {
-                commandManager.removeMapping(registered);
-            }
+        // NOOP: Sponge commands are registered via the event below
+    }
 
-            Optional<CommandMapping> register = commandManager.register(this, new SpongeCommand(runnableFactory, system.getErrorLogger(), command), name);
-            register.ifPresent(commandMapping -> commands.put(name, commandMapping));
-        }
+    @Listener
+    public void onRegisterCommand(RegisterCommandEvent<Command.Raw> event) {
+        // TODO(vankka): component not available here, yet
+//        Subcommand command = component.planCommand().build();
+//        List<String> aliases = new ArrayList<>(command.getAliases());
+//        event.register(plugin, new SpongeCommand(runnableFactory, system.getErrorLogger(), command), aliases.remove(0), aliases.toArray(new String[0]));
```

Comment:
Still haven't figured out a decent way to do this

Reply:
Is `Game` is only available during enable (not load)

If it is only available during enable, the dagger-injected instances could be replaced with a getter, if it is then the component creation could be moved to load (I assume commands are registered after loading)

Reply:
Game itself is but `Server` (`Game#server`) is not, so modules like SpongeSensor and SpongeServerProperties throw a error if the dagger component is initialized onLoad. The command registration happens after onLoad but before onEnable/the `Server` object being ready. 

I'm not very familiar with dagger, pointers to how what you said could be done would be appreciated

Reply:
Ok I think best approach would be to try move any use of `Game#server` to later than construction.

Basically all the `@Inject`, `@Binds` and `@Provides` methods are called when component is built so if possible I would replace any uses `Server` with Game and call Game#server where server was used directly before

Reply:
Righto so the problem was mostly that the config is required to make the command but I was trying to enable the entire PlanSystem, I solved that by only enabling the config on load and the rest on enable. 

The `SpongeServerProperties` still makes the component initialization fails as it relies on using `Game#server` in it's constructor. Otherwise everything will initialize properly

Reply:
Solved over conversation on Discord. Port and max players will be left out as they are unused and cannot be provide without erroring out.